packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts

```diff
@@ -280,79 +305,298 @@ describeNoCompat("GC inactive nodes tests", (getTestObjectProvider) => {
+						code: number;
+						underlyingResponseHeaders?: {
+							[InactiveResponseHeaderKey]: boolean;
+						};
+				  })
+				| undefined;
+
+			/**
+			 * 1. With option, handle.get, throws error with known flag
+			 * 2. With option, with header, ok
+			 * 3. Without option, handle.get, ok
+			 *
+			 * Bonus:
+			 * 1. Summarizer never throws
+			 */
+
+			itExpects(
+				"throwOnInactiveLoad: true; handle.get -- throws and logs",
+				[
+					{
+						eventName:
+							"fluid:telemetry:ContainerRuntime:GarbageCollector:InactiveObject_Loaded",
+					},
+				],
+				async () => {
+					// Create a summarizer client that will be used to summarize the container.
+					const { summarizer: summarizer1 } = await createSummarizer(
+						provider,
+						mainContainer,
+						{
+							runtimeOptions: {
+								gcOptions: { inactiveTimeoutMs },
+							},
+						},
+					);
+
+					// Create a data store, mark it as referenced and then unreferenced
+					const dataStore = await requestFluidObject<ITestDataObject>(
+						await containerRuntime.createDataStore(TestDataObjectType),
+						"",
+					);
+					const url = dataStore.handle.absolutePath;
+					const unreferencedId = dataStore._context.id;
+					defaultDataStore._root.set("dataStore", dataStore.handle);
+					defaultDataStore._root.delete("dataStore");
+
+					// Summarize the container while it's unreferenced. This summary will be used to load another container.
+					const summaryVersion1 = await waitForSummary(summarizer1);
+
+					// Wait for inactive timeout. This will ensure that the unreferenced data store is inactive.
+					await waitForInactiveTimeout();
+
+					// Load a non-summarizer container from the above summary that uses the mock logger. This container has to
+					// be in "write" mode for GC to initialize unreferenced nodes from summary.
+					const container2 = await provider.loadTestContainer(
+						{
+							...testContainerConfigWithThrowOption,
+							loaderProps: { logger: mockLogger },
+						},
+						{ [LoaderHeader.version]: summaryVersion1 },
+					);
+					const defaultDataStoreContainer2 = await requestFluidObject<ITestDataObject>(
+						container2,
+						"/",
+					);
+					defaultDataStoreContainer2._root.set("mode", "write");
+					await waitForContainerWriteModeConnectionWrite(container2);
+
+					// Load the inactive data store. This should result in a loaded event from the non-summarizer container.
+					const handle = manufactureHandle<ITestDataObject>(
+						defaultDataStoreContainer2._context.IFluidHandleContext, // yields the ContaineRuntime's handleContext
+						unreferencedId,
+					);
+					try {
+						// This throws because the DataStore is inactive and throwOnInactiveLoad is set
+						await handle.get();
+						assert.fail("Expected handle.get to throw");
+					} catch (error: any) {
+						const inactiveError: InactiveLoadError = error;
+						assert.equal(inactiveError?.code ?? -1, 404, "Incorrect error status code");
```

Comment:
```suggestion
						assert.equal(inactiveError?.code, 404, "Incorrect error status code");
```
This should work right? Any reason to use -1?

Reply:
Same for the next assert.

Reply:
Mocha has this annoying thing I've noticed where if the type of the args don't match it won't print the actual/expected values when it fails. This is how I workaround it, mostly useful when first developing so I can revert it - seeing status -1 would be confusing if you saw this text fail later and didn't know this code

Reply:
Huh, I never realized that. I always use `assert.strictEqual` though. You can try that maybe.

Reply:
This is using strictEqual, by importing `strict as assert` at the top.  Try it sometime - do `assert.equal(undefined, 123)` and compare the output with `assert.equal(0, 123)`